src/trues.jl

```diff
@@ -0,0 +1,37 @@
+
+"""
+    Trues = Ones{Bool, N, Axes} where {N, Axes}
+
+Lazy version of `trues` with axes.
+Typically created using `Trues(dims)` or `Trues(dims...)`
+
+# Example
+```jldoctest
+julia> Trues(1,3)
+1×3 Ones{Bool,2,Tuple{Base.OneTo{Int64},Base.OneTo{Int64}}} = true
+
+julia> Trues((2,3))
+2×3 Ones{Bool,2,Tuple{Base.OneTo{Int64},Base.OneTo{Int64}}} = true
+```
+"""
+const Trues = Ones{Bool, N, Axes} where {N, Axes}
+
+
+""" `Falses = Zeros{Bool, N, Axes}` (see `Trues`) """
+const Falses = Zeros{Bool, N, Axes} where {N, Axes}
+
+
+# y[mask] = x when mask isa Trues (cf y[:] = x)
+function Base.setindex!(y::AbstractArray{T,N}, x, mask::Trues{N}) where {T,N}
+    @boundscheck size(x) == size(mask) == size(y) || throw(DimensionMismatch())
```

Comment:
Need to put a string description in `DimensionMismatch`. Also, perhaps `axes(x) == axes(mask) == axes(y)` is more general?

Reply:
✓
(found a few others too...)